securepy/stdio.py

```diff
@@ -175,20 +198,23 @@ def restore_std(self) -> None:
             sys.stdout = self.old_stdout
         if isinstance(sys.stderr, LimitedStringIO):
             sys.stderr = self.old_stderr
+        if isinstance(sys.stdin, StringIO):
+            sys.stdin = self.old_stdin
 
     def reset(self) -> None:
         """Reset stored captured stdout & stderr strings."""
-        self.capturing_stdout = LimitedStringIO(self.memory_limit)
-        self.capturing_stderr = LimitedStringIO(self.memory_limit)
+        self.stdout_funnel = LimitedStringIO(self.memory_limit)
+        self.stderr_funnel = LimitedStringIO(self.memory_limit)
 
     def __repr__(self) -> str:
-        return f"<StdCapture(stdout={self.stdout}, stderr={self.stderr})"
+        return f"<IOCage(stdout={self.stdout}, stderr={self.stderr})"
 
 
 def read_process_std(
     process: subprocess.Popen,
     read_chunk_size: int,
     max_size: int,
+    stdin: t.Optional[str] = None,
```

Comment:
I don't think this argument should be here as there's no functionality implemented for sending stdin to process, even though it will be added in the future, there's no reason for an empty argument without any use for now

Reply:
That makes sense and I was planning to add it in later

Reply:
Fixed